internal/nginx/templates/nginx-plus.ingress.tmpl

```diff
@@ -55,21 +55,34 @@ server {
 
 	{{if $server.SSL}}
 	{{if not $server.GRPCOnly}}
-	{{- if $server.SSLRedirect}}
-	if ($scheme = http) {
-		return 301 https://$host:{{index $server.SSLPorts 0}}$request_uri;
-	}
-	{{- end}}
 	{{- if $server.HSTS}}
-	add_header Strict-Transport-Security "max-age={{$server.HSTSMaxAge}}; {{if $server.HSTSIncludeSubdomains}}includeSubDomains; {{end}}preload" always;{{end}}
+	set $hsts_header_val "";
+	proxy_hide_header Strict-Transport-Security;
+	{{- if $server.HSTSBehindProxy}}
+	if ($http_x_forwarded_proto = 'https') {
+	{{else}}
+	if ($https = on) {
+	{{- end}}
+		set $hsts_header_val "max-age={{$server.HSTSMaxAge}}; {{if $server.HSTSIncludeSubdomains}}includeSubDomains; {{end}}preload";
+	}
+
+	add_header Strict-Transport-Security "$hsts_header_val" always;
+	{{end}}
 	{{end}}
 	{{- end}}
+
 	{{- if $server.RedirectToHTTPS}}
 	if ($http_x_forwarded_proto = 'http') {
 		return 301 https://$host$request_uri;
 	}
 	{{- end}}
 
+	{{- if $server.SSLRedirect}}
```

Comment:
the SSL redirect part here must go into the following block
```
	{{if $server.SSL}}
	{{if not $server.GRPCOnly}}
          <here, after hsts stuff>
        {{end}}
        {{end}}
```
Otherwise it is a BUG, because the redirect will be enabled for Ingress resources without any TLS termination.

Please double check that for an Ingress  resource without TLS termination, no SSL redirect is generated.

Reply:
same for nginx oss template

Reply:
👍 nice catch